### PR TITLE
docs: add openldap secret engine

### PR DESCRIPTION
**Pull request by jasonodonnell**
This adds the user guide and API documentation for the OpenLDAP secret engine beta.

### CODE REVIEW

File: website/pages/api-docs/secret/openldap/index.mdx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+| `DELETE` | `/openldap/config` |
+
+This endpoint configures the OpenLDAP secret engine to managed user entries.
+
+Note: the OpenLDAP entry used by `config` should have the neccessary privileges 
+to search and change entry passwords in OpenLDAP.
+
+### Parameters
+
+- `binddn` (string, required) - Distinguished name (DN) of object to bind for managing user entries. Example: `cn=vault,ou=Users,dc=hashicorp,dc=com`
```
Comment (tyrannosaurus-becks) [inline website/pages/api-docs/secret/openldap/index.mdx]: Are these copy pasta'd from other LDAP docs? If so, I won't suggest any formatting changes. If they're not copied, I do notice we've been doing fields differently lately, like this:

```
- `public_key` `(string: <required>)` – Lorum ipsum.
```

Comment (jasonodonnell) [reply website/pages/api-docs/secret/openldap/index.mdx]: These are actually from the AD doc, but I can look into the deltas!

Comment (jasonodonnell) [reply website/pages/api-docs/secret/openldap/index.mdx]: Updated!

File: website/data/api-navigation.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ export default [
         ]
       },
       { category: 'nomad' },
+	  { category: 'openldap' },
```
Comment (tyrannosaurus-becks) [inline website/data/api-navigation.js]: Nit, seems like the left brace is aligned vertically for the rest of the items here.

**Pull request merged**